src/views/view_sycl.hpp

```diff
@@ -100,6 +100,14 @@ struct VectorView<
         disp_((strd > 0) ? disp : disp + (size_ - 1) * (-strd)),
         stride_(strd) {}
 
+  SYCL_BLAS_INLINE VectorView(BufferIterator<scalar_t> data, index_t disp,
+                              increment_t strd, index_t size)
+      : data_{data.template get_range_accessor<
+            cl::sycl::access::mode::read_write>()},
```

Comment:
Basically, are we are specializing the `VectorView` for storing the accessor rather than the buffer with this PR?
So `data_` will contain an accessor that will be in any case `read_write`.

Reply:
I actually had to add this new constructor in this PR to make sure that we are able to allocate the right memory (usm or buffer) without adding anything extra in the `sycl_blas_helper.h` file. The only reason for having this constructor here is to manage any intermediate memory allocation performed by the operator. This constructor is not in any of the `tests` or `benchmarks` or `samples`, but it is required to make the code independent of the helper functions.

Reply:
I was able to remove this constructor altogether by making the changes we discussed earlier.